bowtie/_cli.py

```diff
@@ -49,6 +49,8 @@
     from bowtie._commands import AnyTestResult
     from bowtie._core import DialectRunner, ImplementationInfo, MakeValidator
 
+IMPLEMENTATIONS = Path(__file__).parent.parent / "implementations"
```

Comment:
This is a good thought, but won't work -- in particular it relies on having a checkout of Bowtie present! It won't work once Bowtie is installed. It's unfortunate CI doesn't tell you that, the reason for that is that it obviously runs with a checkout present. Really we should have a separate job that would tell you that this would all be very broken if someone ran `pip install bowtie-json-schema` and didn't also have a checkout.

The way I'd hoped to do this was to use the GitHub packages API to list known implementations -- but unfortunately, it requires authentication (meaning everyone who uses Bowtie would need to login to  use it).

I'm still tied up with some other things so can't give more specific guidance quite yet, but if you want to tackle this issue (which also would be great!) the thing you need to do is learn how to have [hatchling](https://hatch.pypa.io/latest/) compile this data at package build time.

In other words -- when we build Bowtie as part of a release, at *that* time we should essentially compile a list of then-known implementations as a JSON file, and put that within the package.

Alternatively/in addition, we can make use of your `implementations.json` file as well once we have it -- but I think *both* having a local copy and then falling back to retrieve the remote one would be a good long term behavior.

Reply:
Okay @Julian I got your reasoning. I will look into `hatchling` so we have the JSON file during bowtie build process and yes we certainly have `implementations.json` as well (once it is available) to fall back on so I will code that logic as well in case the build JSON file was absent for some reason.

Reply:
Also if we manage to build this JSON file during `hatchling build` process do we still need to start all the individual Docker containers just to get to know the output of `bowtie filter-implementations` ? Seems quite expensive given that we are not running any tests and already have a JSON file from build.

Reply:
We indeed wouldn't! We could put all we need in there.

Reply:
Hey @Julian is it possible to start a bowtie subprocess like how we do in `test_integration.py` to run this command 
`bowtie info $(ls implementations/ | sed 's/^/-i /') --format json > implementations.json` (programmatically obviously)  in the `noxfile.py`'s below session function to guarantee having an `implementations.json` file in the packaged bowtie app? or do we have to force include the file in `pyproject.toml` ?

```python
@session(tags=["build"])
def build(session):
...
```

Reply:
No the `noxfile` isn't the right place, nor is `pyproject.toml` as that will rely on you having a static copy of the file -- I believe the thing you want to look at is [this](https://hatch.pypa.io/1.9/config/build/#build-hooks) -- a build hook plugin which will do this work while hatch(ling) is building the package. That hook should *generate* that file and include it.

Reply:
Okay @Julian I will look into it and revert back if I face some other query.